src/components/OptionListContextProvider.tsx

```diff
@@ -71,6 +77,31 @@ function OptionsListContextProvider({reports, children}: OptionsListProviderProp
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [reports]);
 
+    /**
+     * This effect is used to add a new report option to the list of options when a new report is added to the collection.
+     */
+    useEffect(() => {
+        if (!areOptionsInitialized.current || !reports) {
+            return;
+        }
+        const missingReportId = Object.keys(reports).find((key) => prevReports && !(key in prevReports));
+        const report = missingReportId ? reports[missingReportId] : null;
+        if (!missingReportId || !report) {
```

Comment:
What if we merge collection of reports so there would be more reports missing in that single update, can that happen? @TMisiukiewicz

Reply:
@mountiny that's a valid point, however I couldn't find a quick way how could I test it with many reports, are you aware of any way how to do that? Are merge calls are batched? If so, I assume there could be more than just a single ID missing. I can create a follow-up PR moving from `find` to `filter`